File: abc135/abc135_b.cs
// https://atcoder.jp/contests/abc135/submissions/8233142

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var p = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var result = 0;
            for (int i = 0; i < N; i++)
            {
                if (p[i] != i + 1)
                {
                    result++;
                }
            }
            Console.WriteLine(result <= 2 ? "YES" : "NO");
        }
    }
}

File: abc/abc135/abc135_b.fs
// https://atcoder.jp/contests/abc135/submissions/11957570

let N = stdin.ReadLine() |> int
let p = stdin.ReadLine().Split() |> Array.map int

let rec solver s n =
    if n = N then s
    elif p.[n] <> n + 1 then solver (s + 1) (n + 1)
    else solver s (n + 1)

if solver 0 0 < 3 then "YES"
else "NO"
|> stdout.WriteLine
